src/field/legend_cutscene_camera.c

```diff
@@ -0,0 +1,987 @@
+
+static void startBellShimmer(FieldSystem *fieldSystem);
+static BOOL Task_WaitShimmerEffectAndRestart(TaskManager *taskman);
+static void ClearBellCutscene_CreateField3dObjectTask(ClearBellCutsceneCamera *cam);
+static void Field3dObjectTaskInit_ClearBellCutscene(Field3dObjectTask *task, FieldSystem *fieldSystem, void *taskData);
+static void Field3dObjectTaskDestroy_ClearBellCutscene(Field3dObjectTask *task, FieldSystem *fieldSystem, void *taskData);
+static void Field3dObjectTaskUpdate_ClearBellCutscene(Field3dObjectTask *task, FieldSystem *fieldSystem, void *taskData);
+static void Field3dObjectTaskRender_ClearBellCutscene(Field3dObjectTask *task, FieldSystem *fieldSystem, void *taskData);
+static void bellShimmerReplaceGraphics(ClearBellCutscene3dObjectTaskData *taskData);
+static void modelAnimListSetFrameIndex(Field3DModelAnimation *animation, u32 num, fx32 frame);
+static BOOL modelAnimListAdvanceNoLoop(Field3DModelAnimation *animation, u32 num);
+static void modelAnimListAdvanceLooping(Field3DModelAnimation *animation, u32 num);
+static void setBellsModelsActiveFlag(ClearBellCutscene3dObjectTaskData *taskData, int active);
+static void loadEyeGlimmer3dModel(HeapID heapId, FieldSystem *fieldSystem, LegendCutsceneLugiaEyeGlimmerTaskData *taskData);
+static void unloadEyeGlimmer3dModel(LegendCutsceneLugiaEyeGlimmerTaskData *taskData);
+static BOOL Task_LugiaEyeGlimmer(TaskManager *taskman);
+static BOOL ov02_02251320(TaskManager *taskman);
+static BOOL Task_WaitCameraPan(TaskManager *taskman);
+static BOOL Task_BirdFinalApproach(TaskManager *taskman);
+static void beginWavesEffect(FieldSystem *fieldSystem);
+static void beginLeavesEffect(FieldSystem *fieldSystem);
+static void Task_WavesEffect(SysTask *task, void *taskData);
+static void Task_LeavesEffect(SysTask *task, void *taskData);
+static void endWavesEffect(ClearBellCutscene3dObjectTaskData *taskData);
+static void endLeavesEffect(ClearBellCutscene3dObjectTaskData *taskData);
+static BOOL Task_LugiaArrivesEffectCameraPan(TaskManager *taskman);
+
+static const Field3dObjectTaskTemplate sField3dObjectTaskTemplate_ClearBellCutscene = {
+    3,
+    sizeof(ClearBellCutscene3dObjectTaskData),
+    Field3dObjectTaskInit_ClearBellCutscene,
+    Field3dObjectTaskDestroy_ClearBellCutscene,
+    Field3dObjectTaskUpdate_ClearBellCutscene,
+    Field3dObjectTaskRender_ClearBellCutscene,
+};
+
+static const CameraParam sCameraParam_HeartGold[] = {
+    {
+        .distance = FX32_CONST(666.922119148625),
+        .angle = { .x = 0xF562 },
+        .perspectiveType = 0,
+        .perspective = 0x5C1,
+        .setReference = FALSE,
+    }, {
+        .distance = FX32_CONST(666.922119148625),
+        .angle = { .x = 0xF562 },
+        .perspectiveType = 0,
+        .perspective = 0x5C1,
+        .setReference = FALSE,
+    }, {
+        .distance = FX32_CONST(666.922119148625),
+        .angle = { .x = 0xF762 },
+        .perspectiveType = 0,
+        .perspective = 0xCC1,
```

Comment:
what does perspective refer to and why is is 0xCC1?

Reply:
not sure